library/cwm/src/lib/cwm/dialog.rb

```diff
@@ -0,0 +1,104 @@
+require "yast"
+
+Yast.import "CWM"
+
+module CWM
+  # An OOP API and the pieces missing from {Yast::CWMClass#show Yast::CWM.show}:
+  # - creating and closing a wizard dialog
+  # - Back/Abort/Next buttons
+  #
+  # @see UI::Dialog
+  # @see CWM::AbstractWidget
+  class Dialog
+    include Yast::Logger
+    include Yast::I18n
+    include Yast::UIShortcuts
+
+    # @return [String,nil] The dialog title. `nil`: keep the existing title.
+    def title
+      nil
+    end
+
+    # @return [CWM::WidgetTerm]
+    abstract_method :contents
+
+    # A shortcut for `.new(*args).run`
+    def self.run(*args)
+      new(*args).run
+    end
+
+    # The entry point.
+    # Will open (and close) a wizard dialog unless one already exists.
+    # @return [Symbol]
+    def run
+      if should_open_dialog?
+        wizard_create_dialog { cwm_show }
+      else
+        cwm_show
+      end
+    end
+
+    def should_open_dialog?
+      !Yast::Wizard.IsWizardDialog
+    end
+
+    # The :back button
+    # @return [String,true,nil] button label,
+    #   `true` to use the default label, or `nil` to omit the button
+    def back_button
+      true
+    end
+
+    # The :abort button
+    # @return [String,true,nil] button label,
+    #   `true` to use the default label, or `nil` to omit the button
+    def abort_button
+      true
+    end
+
+    # The :next button
+    # @return [String,true,nil] button label,
+    #   `true` to use the default label, or `nil` to omit the button
+    def next_button
+      true
+    end
+
+    # @return [Array<Symbol>]
+    #   Events for which `store` won't be called, see {Yast::CWMClass#show}
+    def skip_store_for
```

Comment:
question is for which actions it should be done. Currently only use case I see is :redraw. So maybe this should be redraw_event and also handle it in run? Just idea as @teclator currently solving something similar.